### PR TITLE
Data table tests

### PR DESCRIPTION
**Pull request by gijskant**
Refactor data table functionality; add tests.

### CODE REVIEW

File: src/app/services/data-table.service.ts [hunk 1]
```diff
@@ -5,11 +5,14 @@ import {Row} from '../models/table-models/row';
 import {ResourceService} from './resource.service';
 import {Col} from '../models/table-models/col';
 import {ConstraintService} from './constraint.service';
-import {HeaderRow} from '../models/table-models/header-row';
 
 @Injectable()
 export class DataTableService {
 
+  static sortableDimensions = [
```
Comment (beaugogh) [inline src/app/services/data-table.service.ts]: transmart-specific, move it to the transmart-services or transmart-utilities

Comment (beaugogh) [reply src/app/services/data-table.service.ts]: and it seems that these dimensions are not used anywhere?

File: src/app/services/resource.service.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ export class ResourceService {
    * @returns {Observable<Study[]>}
    */
   getStudies(): Observable<Study[]> {
-    return this.transmartResourceService.getStudies();
+    return Observable.fromPromise(this.transmartStudiesService.studies);
```
Comment (beaugogh) [inline src/app/services/resource.service.ts]: I would cache studies in transmart resource service directly

File: src/app/services/resource.service.ts [hunk 3]
```diff
@@ -287,33 +290,44 @@ export class ResourceService {
...
     let offset = dataTable.offset;
     let limit = dataTable.limit;
 
+    // Fetch dimensions for the data matching the constraint
     return this.getDimensions(dataTable.constraint)
       .switchMap((transmartStudyDimensions: TransmartStudyDimensions) => {
+        // Merge available dimensions and current table state
         let tableState: TransmartTableState =
-          TransmartMapper.mapStudyDimensionsToTableState(transmartStudyDimensions, dataTable);
+          TransmartDataTableMapper.mapStudyDimensionsToTableState(transmartStudyDimensions, dataTable);
```
Comment (beaugogh) [inline src/app/services/resource.service.ts]: still doubting the motivation behind separating mappers into even more mappers, if we worry about the transmart mapper getting large, we don't have to be. the mapper functions are just stand-alone, static functions, distinguishing them with comment lines would be enough. Ideally would prefer to keep all the mapper function in one place.

File: src/app/services/resource.service.ts (hunk 2 above)
Comment (beaugogh) [reply src/app/services/resource.service.ts]: also, ideally glowing bear knows one backend-mapper and one backend.resource.service, we should hide backend-specific details from glowing bear as much as possible

**Comment by beaugogh** [comment]
Just a general remark: when we add transmart-specific code, try to isolate such code from the 'glowing bear' code as much as possible, the principle that we try to follow is -- "glowing bear is ignorant about how transmart runs"

File: src/app/services/resource.service.ts (hunk 3 above)
Comment (gijskant) [reply src/app/services/resource.service.ts]: That is not a proper way to organise your code.

Comment (beaugogh) [reply src/app/services/resource.service.ts]: or find alternative ways to hide these other mappers from resource service

**Comment by gijskant** [comment]
@beaugogh Okay now?

**Comment by beaugogh** [comment]
well, still a bit hesitant about the mapper thingy, but will merge

**Pull request merged**